### DISCUSSION TITLE
Question on overriding Paper border color with !important

### DISCUSSION THREAD

**Discussion by mgramigna** [mantinedev/mantine#4333] [Q&A] [answered]
### Description

I'd like to customize the color of the border on a `Paper` component based on the value of a boolean prop passed in to the component that renders the `Paper`. I need access to both `theme` and the value of the prop, so I tried using `sx` on the component. but it requires `!important` to apply the styles. I am just checking if this is the recommended/required way to do this, or if I am missing something.

```tsx
    // assume `selected` is a boolean passed in to the parent component of this Paper
    <Paper
      p="xl"
      withBorder
      sx={(theme) => ({
        // Just trying to set the border color with sx prop doesn't work
        ...(selected && { borderColor: theme.colors.blue[5] })
      })}
      // sx={(theme) => ({
      //   // Setting border color with !important works
      //   ...(selected && { borderColor: `${theme.colors.blue[5]} !important` })
      // })}
    >
      {selected
        ? "Should have a theme blue border"
        : "Should have the default border"}
    </Paper>
```

* I thought about `createStyles`, but can't do that outside the component since it depends on the value of `selected`
* I wanted to avoid using a direct `style` prop on the `Paper` component along with a `useMantineTheme` to get the theme object since this didn't feel like the best way (but maybe it is?)
* Couldn't use `styles` prop since not present on a `Paper` component

### Sandbox Replication

https://codesandbox.io/s/billowing-leaf-vqlbuh?file=/src/PaperExample.tsx

**Comment by rtivital** [comment, marked as answer]
https://codesandbox.io/s/stupefied-pascal-er2k8b?file=/src/PaperExample.tsx

**Reply by mgramigna** [reply]
Perfect, thank you!
